solvers/NonlinearProgram.m

```diff
@@ -155,7 +155,6 @@
       obj.bbcon_lb = [];
       obj.bbcon_ub = [];
       
-      obj = obj.setSolver('default');
```

Comment:
I don't see the need for doing that, or I didn't understand you.
Look at line 734 in NonlinearProgram.  Setting to 'default' causes setSolver to run through a series of if-elseif that look for a solver in some priority order, and in particular, for checking studentSnopt, the number of variables and constraints have to be already set.
You see the motivation in the comment I left on the commit. I'll paraphrase it here for you:
Run runDircolPerching.m having studentSnopt (and not the full Snopt) installed.  Make sure that N=41 or higher.  You'll understand when you see the error it throws.  I wrote it in the comment I left.  Hope this helps.